src/main/java/fruitshop/service/data/ProcessDataService.java

```diff
@@ -0,0 +1,11 @@
+package fruitshop.service.data;
+
+import fruitshop.model.FruitTransaction;
+import fruitshop.model.Operation;
+import fruitshop.strategy.StrategyService;
+import java.util.List;
+import java.util.Map;
+
+public interface ProcessDataService {
+    void process(List<FruitTransaction> list, Map<Operation, StrategyService> operationsMap);
+}
```

Comment:
instead of passing operationMap to this method, it is better to create OperationStrategy class that will just contain this map that should be passed using constructor and method get() that will return specific StrategyService (it's better to rename it to OperationHandler)

when you do this you don't have to pass this map here and you can just create an instance of a strategy inside the method and get the needed StrategyService to process the transaction

Reply:
understood